generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java

```diff
@@ -90,4 +90,192 @@ public void testInvalidNullDefaultForPotentiallyNonNullableArray() throws Except
                     e.getMessage().contains("not all versions of this field are nullable"));
         }
     }
+
+    @Test
+    public void testInvalidFieldName() {
+        try {
+            MessageGenerator.JSON_SERDE.readValue(String.join("", Arrays.asList(
+                "{",
+                "  \"type\": \"request\",",
+                "  \"name\": \"FooBar\",",
+                "  \"validVersions\": \"0-2\",",
+                "  \"fields\": [",
+                "    { \"name\": \"_badName\", \"type\": \"[]int32\", \"versions\": \"0+\" }",
+                "  ]",
+                "}")), MessageSpec.class);
+            fail("Expected MessageDataGenerator constructor to fail");
```

Comment:
I would suggest using `assertThrows` in this and other tests added in this PR that validate that an exception is thrown.